github/pull_request.go

```diff
@@ -153,29 +158,49 @@ func patchToLineBounds(patch string) ([]lineBound, error) {
 	scanner := bufio.NewScanner(strings.NewReader(patch))
 	for scanner.Scan() {
 		line := scanner.Text()
-		// patch header lines are formatted like:
-		// @@ -0,0 +1,5 @@ <arbitrary line of code which may be blank>
-		if len(line) >= 15 && strings.HasPrefix(line, "@@ -") && strings.Contains(line[2:], " @@") {
-			// split into four pieces: (0) @@, (1) old line number and offset, (2), new line number and offset, (3) @@
-			segments := strings.Split(line, " ")
-			if len(segments) < 4 {
-				continue
-			}
+		match := namedMatch(diffRangeRegexp, line)
+		if match == nil {
+			continue
+		}
 
-			// split and parse new line numbers
-			bounds := strings.Split(segments[2], ",")
-			start, err := strconv.Atoi(bounds[0][1:]) // drop the "+"
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to convert %v to integer while processing %q", bounds[0][1:], line)
-			}
-			endOffset, err := strconv.Atoi(bounds[1]) // one-indexed offset (subtract 1 when using)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to convert %v to integer while processing %q", bounds[0][1:], line)
+		// NOTE: each of these named matches must have integers
+		// because the regexp specifically checks for `\d+`.
+		oldStart, _ := strconv.Atoi(match["old_start"])
+		oldRows, _ := strconv.Atoi(match["old_rows"])
+		newStart, _ := strconv.Atoi(match["new_start"])
+		newRows, _ := strconv.Atoi(match["new_rows"]) // one-indexed offset (subtract 1 when using)
+
+		rowDiff := newRows - oldRows
+		for i, bound := range lineBounds {
+			if oldStart < bound.start {
+				bound.start += rowDiff
+				bound.end += rowDiff
+			} else if oldStart < bound.end {
+				bound.end += rowDiff
 			}
-
-			lineBounds = append(lineBounds, lineBound{start: start, end: start + endOffset - 1})
+			lineBounds[i] = bound
```

Comment:
I don't like the asymptotic runtime of this little bit right here. Let me know if you expect this to be a problem and I can come back and do it right.